### DISCUSSION TITLE
Reset Date Range Picker

### DISCUSSION THREAD

**Discussion by eliobricenov** [adobe/react-spectrum#3318] [Q&A]
Is there any way to reset the selected range? I've looked everywhere and couldn't find anything related.

For example, in this demo https://codesandbox.io/s/objective-shape-8r4utm?file=/src/RangeCalendar.js I'd like to add a clear button that'd reset the state back to its placeholder value.

**Comment by reidbarber** [comment]
You can use `state.setValue()` (where state is from `useRangeCalendarState` ) to set the state to the same value as the default value.

Here is a forked version of the example you linked, but with a clear button added: https://codesandbox.io/s/daterangepicker-with-clear-button-13l5s6?file=/src/RangeCalendar.js

**Reply by eliobricenov** [reply]
Thanks for answering! 

Actually what I really want to do is to clear the state values, this means resetting the range to having no values.  In this fork, you can see that the initial range is not set, with the reset button I'd lo return to this state. However all the methods to change the state need `start` and `end`, I'd probably need something like `state.reset()`

**Reply by reidbarber** [reply]
In that case, you can do `state.setValue(null)`. I just updated the sandbox.

**Reply by eliobricenov** [reply]
Hmm, in your project the button works but I'm using NextJS with Typescript in mine so I get this error: 

"TypeError: can't access property "start", range is null"

<img width="908" alt="image" src="https://user-images.githubusercontent.com/38574891/180367540-595127c5-e54f-45e0-83c6-2c65d7f9d188.png">

**Reply by eliobricenov** [reply]
let me try to get a sandbox for you

**Reply by eliobricenov** [reply]
hmmm, idk in this fork I can clear https://codesandbox.io/s/quizzical-wildflower-ngomh6?file=/src/App.tsx idk what's happening :/, maybe is Next JS?

**Reply by eliobricenov** [reply]
well, this seems to work too: https://codesandbox.io/s/recursing-ully-szzi9p

I'll have to debug further and report back.

Thanks a lot for the help, appreciate it, awesome work with this library.

**Reply by eliobricenov** [reply]
Ok, so it seems that I get the error because I'm also using `useDateRangePicker` hook, and setting the state as null is breaking something somewhere.

**Reply by eliobricenov** [reply]
Ok I finally was able to create a sandbox with the error: https://codesandbox.io/s/boring-dust-mpcbgk?file=/src/RangeCalendar.js

**Comment by eliobricenov** [comment]
Sorry for the previous spam. Ultimately, the error can be reproduced here:
> Ok I finally was able to create a sandbox with the error: https://codesandbox.io/s/boring-dust-mpcbgk?file=/src/RangeCalendar.js
it seems that I get the error because I'm also using `useDateRangePicker` hook, and setting the state as null is breaking something somewhere.

Do you have anything in mind? @reidbarber

**Reply by reidbarber** [reply]
This actually looks like a bug. I think we're missing an optional chaining operator here: https://github.com/adobe/react-spectrum/blob/92294d1271e49d55e82f30067a388d4ff2973421/packages/%40react-stately/datepicker/src/useDateRangePickerState.ts#L80
